pkg/network/status/status.go

```diff
@@ -395,3 +394,37 @@ func (m *Prober) updateStates(ingressState *ingressState, podState *podState) {
 		}
 	}
 }
+
+func (m *Prober) probeVerifier(item *workItem) prober.Verifier {
+	return func(r *http.Response, _ []byte) (bool, error) {
+		// In the happy path, the probe request is forwarded to Activator or Queue-Proxy and the response (HTTP 200)
+		// contains the "K-Network-Hash" header that can be compared with the expected hash. If the hashes match,
+		// probing is successful, if they don't match, a new probe will be sent later.
+		// An HTTP 404 is expected in the case of the creation of a new Knative service because the rules will
+		// not be present in the Envoy config until the new VirtualService is applied.
+		// No information can be extracted from any other scenario (e.g. HTTP 302), therefore in that case,
+		// probing is assumed to be successful because it is better to say that an Ingress is Ready before it
+		// actually is Ready than never marking it as Ready. It is best effort.
+		switch r.StatusCode {
+		case http.StatusOK:
+			hash := r.Header.Get(network.HashHeaderName)
+			switch hash {
+			case "":
```

Comment:
Please add some documentation regarding this case?  The documentation on #402 only says that if hashes don't match a new probe will be sent later, which is not the case here.

Reply:
The comment says `In the happy path ... the response (HTTP 200) contains the "K-Network-Hash" header`, so here is the case where the header is **not present**.

Reply:
I am going to change that to r.Header.Values(network.HashHeaderName)

Reply:
Header.Values is only in Go 1.14, so I can't use it for now.